python/mxnet/gluon/utils.py

```diff
@@ -56,13 +56,10 @@ def split_data(data, num_slice, batch_axis=0, even_split=True):
     Returns
     -------
     list of NDArray
-        Return value is a list even if `num_slice` is 1.
+        Return value is a list even if `num_slice` is 1. When `even_split`
+        is `False` this may be shorter than `num_slice`.
```

Comment:
Is this a good idea?
when used for multi-gpu, this would cause some gpus to not get data, then gradient will be stale during update

Reply:
See comment above: A user training using `zip(context, slices)` then doing`L.backward()` inside the loop won't have problems.

Also, suppose a user doesn't use this idiom and is unlucky (has `N` samples where `N % batch_size < num_slice`). At worst, MXNet warns them about the stale gradient; plus, it would occur on the very last batch. This is still preferable to doing training/inference just to error on the last batch with a `size < num_slice` message, requiring an ad-hoc workaround.

Reply:
I think it will fail at step due to stale gradient

Reply:
It's better to use the discard or rollover option for DataLoader